lib/internal/modules/esm/loader.js

```diff
@@ -107,6 +109,15 @@ class Loader {
     }
 
     const { format } = getFormatResponse;
+    if (format === null) {
```

Comment:
A cursory look at this makes me wonder if we know that `format` destructured from `getFormatResponse` (via `this._getFormat()`) is `null` (guessing this means _no translator_) only in this particular instance. Without digging any deeper right now, would you be able to confirm/deny? Perhaps that could be an oversight. Let me know as I haven't been digging around here in _months_.

Reply:
I haven't encountered `null` value for format except in this use case, but I may have missed something, not sure.

Reply:
Please don't land this until we are able to know for certain that this is the only scenario where we get a `null` format.

I have never encountered a `null` format while playing with the `getFormat` experimental loader hook. 

I am pretty sure this behavior is undocumented and may actually be an internal edge case not to be leveraged this way.

Please hold off until we can investigate further as to why this is `null` as opposed to one of the formats we've defined.

Reply:
This PR doesn’t change the current behavior, it changes the error that’s being reported. I’m all for investigate any undefined behavior, but I don’t think there’s any leveraging enabled by this PR. Why would you hold off? Can’t this land and be improved later?

Reply:
> Why would you hold off?

My current opinion is that this is not the correct way for us to determine if the module specifier is using an unsupported MIME type internally.

> Can’t this land and be improved later?

Yeah, you can actually disregard my previous comment if it suits you. My concern about the way we are checking for a `null` format here may only be able to be addressed in a larger-scale refactoring, which may be out of scope for this PR.

We can always revisit this internal check in a later PR if you would prefer.

Reply:
> My current opinion is that this is not the correct way for us to determine if the module specifier is using an unsupported MIME type internally.

This is probably linked to the current loader implementation, something to have in mind when refactoring it.